wallet/reference/non-evm-apis/starknet-snap-api.md

```diff
@@ -0,0 +1,830 @@
+---
+description: See the Starknet Snap API reference.
+sidebar_position: 1
+toc_max_heading_level: 2
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Starknet Snap API
+
+With the [Starknet Snap](../../how-to/use-non-evm-networks/starknet/index.md) installed, dapps can
+use the Starknet Snap API to interact with users' Starknet accounts (for example, to send transactions).
+
+:::note
+You can also communicate with the Starknet network using the
+[Starknet API](/services/reference/starknet).
+:::
+
+## `starkNet_createAccount`
+
+Deploys an account contract.
+
+### Parameters
+
+- `addressIndex`: `integer` - (Optional) Specific address index of the derived key in
+  [BIP-44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki).
+- `deploy`: `boolean` - (Optional) Indicate whether to include send the deploy transaction for the
+  account contract.
+  The default is `false`.
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Goerli testnet.
+
+### Returns
+
+The response from Starknet's `gateway/add_transaction` API endpoint.
+
+### Example
+
+<Tabs>
+<TabItem value="Request">
+
+```js
+await window.ethereum.request({
+  method: "wallet_invokeSnap",
+  params: {
+    snapId: "npm:@consensys/starknet-snap",
+    request: {
+      method: "starkNet_createAccount",
+      "params": {
+        "addressIndex": 1,
+        "deploy": true,
+        "chainId": "0x1"
```

Comment:
we could provide here an existing chainId, maybe the testnet one ? Or make clear what the chain id should be at the beginning of the page.